experimental/oauther/oauth2.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package oauther
+
+import (
+	"encoding/json"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"golang.org/x/oauth2"
+)
+
+const (
+	oAuth2StateTimeToLive = 300 // seconds
+)
+
+type OAuther interface {
+	GetToken(userID string) (*oauth2.Token, error)
+	GetURL() string
+	Deauth(userID string) error
+}
+
+type oAuther struct {
+	api             plugin.API
+	onConnect       func(userID string, token *oauth2.Token)
+	storePrefix     string
+	pluginURL       string
+	oAuthURL        string
+	connectedString string
+	config          *oauth2.Config
+	logger          logger.Logger
+}
+
+func NewOAuther(
+	r *mux.Router,
+	api plugin.API,
+	pluginURL,
```

Comment:
This seems like something the component can figure out for itself, vs. needing to be set.

Reply:
If we pass the whole plugin.API, yes, it is something it can figure out itself. Nevertheless, I think that passing the whole api is not the correct approach, specially since we only want the API to access the KVStore. I changed this to pass a KVStore interface (that should be met by the StoreService) and remove the plugin.API reference.